components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -650,22 +651,28 @@ public static void validateAppliedPolicy(OperationPolicy appliedPolicy,
 
         if (policySpec == null && apiUUID != null) {
             // if policy is not found in the project, policy can be referenced from an existing policy.
-            OperationPolicyData policyData =
-                    provider.getAPISpecificOperationPolicyByPolicyName(appliedPolicy.getPolicyName(),
-                            appliedPolicy.getPolicyVersion(), apiUUID, null, tenantDomain, false);
-            if (policyData != null) {
-                policySpec = policyData.getSpecification();
-                appliedPolicy.setPolicyId(policyData.getPolicyId());
+            if (policyType == null || ImportExportConstants.POLICY_TYPE_API.equalsIgnoreCase(policyType)) {
+                // if policy type is 'api' or not specified, then search API specific operation policies
+                OperationPolicyData policyData =
+                        provider.getAPISpecificOperationPolicyByPolicyName(appliedPolicy.getPolicyName(),
+                                appliedPolicy.getPolicyVersion(), apiUUID, null, tenantDomain, false);
+                if (policyData != null) {
+                    policySpec = policyData.getSpecification();
+                    appliedPolicy.setPolicyId(policyData.getPolicyId());
+                }
             }
         }
 
         if (policySpec == null) {
-            OperationPolicyData policyData =
-                    provider.getCommonOperationPolicyByPolicyName(appliedPolicy.getPolicyName(),
-                            appliedPolicy.getPolicyVersion(), tenantDomain, false);
-            if (policyData != null) {
-                policySpec = policyData.getSpecification();
-                appliedPolicy.setPolicyId(policyData.getPolicyId());
+            if (policyType == null || ImportExportConstants.POLICY_TYPE_COMMON.equalsIgnoreCase(policyType)) {
```

Comment:
Same here. Why do we have the `null` or `common` check and no else conditions?

Reply:
if policy is not found in the project, policy can be referenced from an existing policy. That logic is performed here.
(When importing an API which has been exported from a previous APIM version which doesn't have policy support, the default policy type is set to null.)
In the process of finding policy among existing policies :
1. check if the policy is within API Specific operational policies. this check is made for the policies which the type is 'api' or not defined which is filtered by null check
2. check if the policy is within Common operational policies. this check is made for the policies which the type is 'common' or not defined which is filtered by null check